rclcpp_examples/CMakeLists.txt

```diff
@@ -101,23 +101,35 @@ if(AMENT_ENABLE_TESTING)
     find_package("${middleware_impl}" REQUIRED)
   endforeach()
 
-  foreach(example_executable list_parameters_async list_parameters parameter_events_async parameter_events set_and_get_parameters_async set_and_get_parameters)
+  # Add each test case.  Multi-executable tests can be specified in
+  # colon-separated strings, like  exe1:exe2.
+  foreach(example_executable list_parameters_async list_parameters parameter_events_async parameter_events set_and_get_parameters_async set_and_get_parameters talker:listener)
+
+    set(RCLCPP_EXAMPLES_EXPECTED_OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/test/${example_executable}")
+    string(
+      REPLACE ":" ":${CMAKE_CURRENT_SOURCE_DIR}/test/"
+      RCLCPP_EXAMPLES_EXPECTED_OUTPUT ${RCLCPP_EXAMPLES_EXPECTED_OUTPUT})
```

Comment:
I've changed the CMake to use a list of lists and use a `foreach` loop instead of `string(REPLACE...)`. Please let me know if that's more readable. Thanks!